### DISCUSSION TITLE
Kotlin?

### DISCUSSION THREAD

**Discussion by trymeouteh** [amir1376/ab-download-manager#14] [General]
This desktop app is fully written in Kotlin?

Can this app also be compilled for MacOS, Android and iOS or is there limitation to Kotlin apps that are not made for Android?

**Comment by amir1376** [comment]
> This desktop app is fully written in Kotlin?

Yes, `kotlin` is used for the entire app

> Can this app also be compilled for MacOS, Android and iOS or is there limitation to Kotlin apps that are not made for Android?

There is no limitation
- You already can compile it for macOS ( I don't have macOS so I can't tell you it works as expected )
- Android needs a little work, some UI pages needs to be responsive to work on mobiles too
- I don't know much about iOS ecosystem, if they allow apps to store and modify files in `Random Access` manner, Yes it can be compiled for iOS too (of course with some extra work)


I plan to also release the **Android** version in the future if the **Desktop** version is well received by the community, but my priority is to make desktop app more stable at the moment

**Reply by ghost** [reply]
> > This desktop app is fully written in Kotlin?
> 
> Yes, `kotlin` is used for the entire app
> 
> > Can this app also be compilled for MacOS, Android and iOS or is there limitation to Kotlin apps that are not made for Android?
> 
> There is no limitation
> 
> * You already can compile it for macOS ( I don't have macOS so I can't tell you it works as expected )
> * Android needs a little work, some UI pages needs to be responsive to work on mobiles too
> * I don't know much about iOS ecosystem, if they allow apps to store and modify files in `Random Access` manner, Yes it can be compiled for iOS too (of course with some extra work)
> 
> I plan to also release the **Android** version in the future if the **Desktop** version is well received by the community, but my priority is to make desktop app more stable at the moment

This is absolutely an OSM project, appreciation in words is almost impossible, still thanks a lot.

**Comment by amir1376** [comment]
Just for an update, the macOS and Android version are ready to use!
